frontend/packages/citizen-frontend/src/decisions/decision-response-page/DecisionResponse.tsx

```diff
@@ -134,16 +136,15 @@ export default React.memo(function DecisionResponse({
                 disabled={blocked || submitting}
                 dataQa={'radio-accept'}
               />
-              <DatePickerDeprecated
-                date={requestedStartDate}
-                onChange={setRequestedStartDate}
-                type="short"
-                minDate={startDate}
-                maxDate={
-                  ['PRESCHOOL', 'PREPARATORY_EDUCATION'].includes(decisionType)
-                    ? startDate
-                    : startDate.addDays(14)
+              <DatePicker
+                date={requestedStartDate.format()}
+                onChange={(date: string) =>
+                  setRequestedStartDate(LocalDate.parseFi(date))
```

Comment:
I don't think this is ok. The type must be string all the way in state.

Reply:
Also there's another ticket that for preschool/preparatory we should not use date picker since changing the date is not allowed anyway. It might make sense to implement that at the same time.

Reply:
Changed the startDate and endDate to be strings all the way from the `deserializeDecision` function. Also changed the startDate to be a string on the page in `PRESCHOOL` and `PREPARATORY_EDUCATION` cases.

Reply:
great! I'll test tomorrow

Reply:
Does't quite work, tested with connected daycare

1. In datepicker all dates look disabled whether valid or not.
2. In datepicker any date can be selected whether valid or not.
3. Any date can be written (as should be), but no validation warning is shown if the date is outside the accepted range, nor is submit button disabled.

Reply:
Fixed issues with users being able to select non-valid dates and added warnings on disallowed dates.